File: app/lib/core/env.dart
/// Compile-time configuration. Supply via --dart-define so secrets never live
/// in source control (Section 10, P0 "API keys never in client code").
///
/// Note: the Supabase anon key is safe to ship in the client — it only grants
/// access through RLS. Service-role keys, Razorpay/Vision/OneSignal secrets
/// live exclusively in Supabase Edge Function env vars and are NEVER bundled
/// in the app.
///
/// Example:
///   flutter run \
///     --dart-define=SUPABASE_URL=https://xxxx.supabase.co \
///     --dart-define=SUPABASE_ANON_KEY=eyJ...
abstract final class Env {
  static const String supabaseUrl =
      String.fromEnvironment('SUPABASE_URL', defaultValue: '');
  static const String supabaseAnonKey =
      String.fromEnvironment('SUPABASE_ANON_KEY', defaultValue: '');

  static const String oneSignalAppId =
      String.fromEnvironment('ONESIGNAL_APP_ID', defaultValue: '');

  static bool get isConfigured =>
      supabaseUrl.isNotEmpty && supabaseAnonKey.isNotEmpty;
}

File: web/src/core/env.ts
// Compile-time configuration via Vite env vars (VITE_ prefix). Secrets for
// server-only integrations (Vision, Razorpay, OneSignal, service-role key)
// live exclusively in Supabase Edge Function env vars — never here (Section 10).
//
// Provide in web/.env.local:
//   VITE_SUPABASE_URL=https://<project>.supabase.co
//   VITE_SUPABASE_ANON_KEY=<anon-key>
//
// The anon key is safe in the client because RLS gates every row.
export const env = {
  supabaseUrl: import.meta.env.VITE_SUPABASE_URL ?? "",
  supabaseAnonKey: import.meta.env.VITE_SUPABASE_ANON_KEY ?? "",
  // Razorpay publishable key id (rzp_test_… / rzp_live_…). Safe in the client;
  // the secret stays server-side (Edge Function webhook verifies payments).
  razorpayKeyId: import.meta.env.VITE_RAZORPAY_KEY_ID ?? "",
};

export const isConfigured = () =>
  env.supabaseUrl.length > 0 && env.supabaseAnonKey.length > 0;
